recipes/genie/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os, glob
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class GenieConan(ConanFile):
+    name = "genie"
+    license = "BSD-3-clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bkaradzic/GENie"
+    description = "Project generator tool"
+    topics = ("conan", "genie", "project", "generator", "build")
+    settings = "os", "arch", "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def build_requirements(self):
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("cccl/1.1")
+
+        if self.settings.os == "Windows" and tools.os_info.is_windows:
+            if "make" not in os.environ.get("CONAN_MAKE_PROGRAM", ""): # or not tools.which("make"):
+                self.build_requires("make/4.2.1")
+
+            if "CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() != 'msys2':
+                self.build_requires("msys2/20200517")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("GENie-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    @property
+    def _os(self):
+        return {
+            "Windows": "windows",
+            "Linux": "linux",
+            "Macos": "darwin",
+            "FreeBSD": "freebsd",
+        }[str(self.settings.os)]
+
+    @property
+    def _cc(self):
```

Comment:
Except for Visual Studio, I think this should also respect `tools.getenv("CC")`.

Reply:
`tools.getenv("CC")` does not respect the conan settings and it may differ from the `self.settings.compiler`.

Reply:
It's the responsibility of the user to set `CC` correctly.
Assume you're cross compiling genie for aarch64.
Then you don't want to use `gcc`, but `aarch64-linux-gnu-gcc`.
Which you can override by using the `CC` environment variable.